Enhance ansible playbooks and cleanup in simple-volume

vijaysinghrawat1 opened the pull request:
Signed-off-by: vijaysinghrawat1 <vijaysinghrawat167@gmail.com>

**What this PR does / why we need it**:
As part of this task move stern logger to pre-req.yml
Run each application on its own ns
Perform apply/delete applications using modularization by passing required arguments.
Cleanup should give out result by setting flag"Cleanup Pass" or "Cleanup Fail"
Slack should notify test result + cleanup result
Remove steps to delete operator in couple of playbooks.
ANSIBLE-LINT check
```
vijay@vijay-virtual-machine:~/openebs/e2e/ansible$ ansible-lint playbooks/feature/snapshots/simple-volume/snapshot.yml 
[ANSIBLE0013] Use shell only when shell functionality is required
playbooks/feature/snapshots/simple-volume/snapshot.yml:87
Task/Handler: Establish iSCSI session with volume
```


**Special notes for your reviewer**:
@dargasudarshan @ksatchit please review.

dargasudarshan commented on e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot-cleanup.yml:
```diff
@@ -4,6 +4,7 @@
     executable: /bin/bash
   delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+  changed_when: true
 
```
@vijaysinghrawat1 Seems like you did not implemented changes for cleanup file?

dargasudarshan commented on e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot-vars.yml:
```diff
@@ -14,4 +18,6 @@ test_data_size: 1
 
 checksum: sha1
 
+ns: openebs
```
this variable is not required

dargasudarshan commented on e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:
```diff
@@ -1,54 +1,42 @@
+         command: kubectl create ns {{ namespace }}
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
 
        - name: Create a storage volume via a pvc
```
Use include task to deploy application

dargasudarshan commented on e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:
```diff
@@ -1,54 +1,42 @@
          delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
 
        - name: Confirm volume container is running
          shell: >
```
Use include task to check status

dargasudarshan commented on e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:
```diff
@@ -247,54 +235,65 @@
+         delegate_to: "{{groups['kubernetes-kubeminions'].0}}"
 
-       - set_fact: 
+       - name: Test Passed
```
Set flag as "Test Passed/Failed"

dargasudarshan commented on e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:
```diff
@@ -247,54 +235,65 @@
+
+        - name: Test Cleanup Pass
+          set_fact:
+            cflag: cleanup_pass
```
Same as previous comment

dargasudarshan commented:
@vijaysinghrawat1 Can you paste test playbook output.

dargasudarshan commented on e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:
```diff
@@ -1,54 +1,41 @@
+       - include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+         vars:
+            ns: openebs
+            app: maya-api
```
Since we are updating the helm to use stable charts, maya-api server is deprecated and now it will be openeb-api server. Can you make the changes accordingly?

vijaysinghrawat1 commented:
```
ansible-playbook 2.5.2
  config file = /home/vijay/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/home/vijay/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /home/vijay/openebs/e2e/ansible/ansible.cfg as config file
statically imported: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot-cleanup.yml
statically imported: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/simple-volume-prerequisites.yml

PLAYBOOK: snapshot.yml ************************************************************************************************************************
1 plays in playbooks/feature/snapshots/simple-volume/snapshot.yml

PLAY [localhost] ******************************************************************************************************************************

TASK [Gathering Facts] ************************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:2
ok: [localhost]
META: ran handlers

TASK [Get $HOME of K8s master for kubernetes user] ********************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/simple-volume-prerequisites.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; echo $HOME", "delta": "0:00:00.004792", "end": "2018-05-24 11:11:26.145063", "rc": 0, "start": "2018-05-24 11:11:26.140271", "stderr": "", "stderr_lines": [], "stdout": "/home/ubuntu", "stdout_lines": ["/home/ubuntu"]}

TASK [Start the log aggregator to capture test pod logs] **************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/simple-volume-prerequisites.yml:10
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; nohup stern \"maya*|openebs*|pvc*\" --since 1m > \"/home/ubuntu/setup/logs/snap_create_revert.log\" &", "delta": "0:00:00.004947", "end": "2018-05-24 11:11:26.597954", "rc": 0, "start": "2018-05-24 11:11:26.593007", "stderr": "mesg: ttyname failed: Inappropriate ioctl for device\n/bin/bash: /home/ubuntu/setup/logs/snap_create_revert.log: No such file or directory", "stderr_lines": ["mesg: ttyname failed: Inappropriate ioctl for device", "/bin/bash: /home/ubuntu/setup/logs/snap_create_revert.log: No such file or directory"], "stdout": "", "stdout_lines": []}

TASK [include_tasks] **************************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:11
included: /home/vijay/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check maya-api pod status] **************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n openebs | grep maya-api", "delta": "0:00:00.416711", "end": "2018-05-24 11:11:27.591158", "rc": 0, "start": "2018-05-24 11:11:27.174447", "stderr": "", "stderr_lines": [], "stdout": "maya-apiserver-6d945596b-ff4ht         1/1       Running   0          10m", "stdout_lines": ["maya-apiserver-6d945596b-ff4ht         1/1       Running   0          10m"]}

TASK [Store maya-apiserver pod name in variable] **********************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:16
ok: [localhost] => {"ansible_facts": {"maya_pod": "maya-apiserver-6d945596b-ff4ht"}, "changed": false}

TASK [Copy the volume claim to kube master] ***************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:20
ok: [localhost -> None] => {"changed": false, "checksum": "ffac0ad06ec57f90a3c7da9d5e626256c9351e0f", "dest": "/home/ubuntu/test-pvc.yaml", "gid": 1000, "group": "ubuntu", "mode": "0664", "owner": "ubuntu", "path": "/home/ubuntu/test-pvc.yaml", "size": 188, "state": "file", "uid": 1000}

TASK [Create test specifi namespace] **********************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:26
changed: [localhost -> None] => {"changed": true, "cmd": ["kubectl", "create", "ns", "simple-volume"], "delta": "0:00:00.240639", "end": "2018-05-24 11:11:29.784036", "rc": 0, "start": "2018-05-24 11:11:29.543397", "stderr": "", "stderr_lines": [], "stdout": "namespace \"simple-volume\" created", "stdout_lines": ["namespace \"simple-volume\" created"]}

TASK [include_tasks] **************************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:30
included: /home/vijay/openebs/e2e/ansible/playbooks/utils/deploy_task.yml for localhost

TASK [Deploy application files] ***************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/utils/deploy_task.yml:2
changed: [localhost -> None] => (item=test-pvc.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f test-pvc.yaml -n simple-volume", "delta": "0:00:00.407716", "end": "2018-05-24 11:11:30.850668", "item": "test-pvc.yaml", "rc": 0, "start": "2018-05-24 11:11:30.442952", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"vut\" created", "stdout_lines": ["persistentvolumeclaim \"vut\" created"]}

TASK [Confirm volume container is running] ****************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:35
FAILED - RETRYING: Confirm volume container is running (10 retries left).
changed: [localhost -> None] => (item=ctrl) => {"attempts": 2, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n simple-volume | grep pvc | grep ctrl | grep Running | wc -l", "delta": "0:00:00.313716", "end": "2018-05-24 11:12:02.433350", "item": "ctrl", "rc": 0, "start": "2018-05-24 11:12:02.119634", "stderr": "", "stderr_lines": [], "stdout": "1", "stdout_lines": ["1"]}
changed: [localhost -> None] => (item=rep) => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n simple-volume | grep pvc | grep rep | grep Running | wc -l", "delta": "0:00:00.355167", "end": "2018-05-24 11:12:03.090732", "item": "rep", "rc": 0, "start": "2018-05-24 11:12:02.735565", "stderr": "", "stderr_lines": [], "stdout": "2", "stdout_lines": ["2"]}

TASK [Get storage ctrl pod name] **************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:50
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get pods -n simple-volume | grep ctrl", "delta": "0:00:00.302582", "end": "2018-05-24 11:12:03.822773", "rc": 0, "start": "2018-05-24 11:12:03.520191", "stderr": "", "stderr_lines": [], "stdout": "pvc-360154c4-5f43-11e8-9b34-02b983f0a4db-ctrl-86784c6466-qqx2z   2/2       Running   0          32s", "stdout_lines": ["pvc-360154c4-5f43-11e8-9b34-02b983f0a4db-ctrl-86784c6466-qqx2z   2/2       Running   0          32s"]}

TASK [Set ctrl pod name to variable] **********************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:57
ok: [localhost] => {"ansible_facts": {"ctrl_pod_name": "pvc-360154c4-5f43-11e8-9b34-02b983f0a4db-ctrl-86784c6466-qqx2z", "vol_name": "pvc-360154c4-5f43-11e8-9b34-02b983f0a4db"}, "changed": false}

TASK [Get IP address of ctrl pod] *************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:62
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl describe pod pvc-360154c4-5f43-11e8-9b34-02b983f0a4db-ctrl-86784c6466-qqx2z -n simple-volume | grep IP", "delta": "0:00:00.336788", "end": "2018-05-24 11:12:04.703619", "rc": 0, "start": "2018-05-24 11:12:04.366831", "stderr": "", "stderr_lines": [], "stdout": "IP:             10.36.0.3\n      --clusterIP", "stdout_lines": ["IP:             10.36.0.3", "      --clusterIP"]}

TASK [Set IP of Pod to variable] **************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:69
ok: [localhost] => {"ansible_facts": {"ctrl_ip": "10.36.0.3"}, "changed": false}

TASK [Establish iSCSI session with volume] ****************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:86
changed: [localhost -> None] => {"changed": true, "cmd": "iscsiadm -m discovery -t st -p 10.36.0.3:3260", "delta": "0:00:00.084407", "end": "2018-05-24 11:12:07.259795", "rc": 0, "start": "2018-05-24 11:12:07.175388", "stderr": "", "stderr_lines": [], "stdout": "10.104.198.194:3260,1 iqn.2016-09.com.openebs.jiva:pvc-360154c4-5f43-11e8-9b34-02b983f0a4db", "stdout_lines": ["10.104.198.194:3260,1 iqn.2016-09.com.openebs.jiva:pvc-360154c4-5f43-11e8-9b34-02b983f0a4db"]}

TASK [Store target iqn in variable] ***********************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:92
ok: [localhost] => {"ansible_facts": {"iqn": "iqn.2016-09.com.openebs.jiva:pvc-360154c4-5f43-11e8-9b34-02b983f0a4db"}, "changed": false}

TASK [Login to iSCSI target] ******************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:96
changed: [localhost -> None] => {"changed": true, "connection_changed": true, "devicenodes": ["/dev/sdc"], "nodes": ["iqn.2016-09.com.openebs.jiva:pvc-9c4b32c0-5a8a-11e8-a847-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-13416471-5e5b-11e8-add9-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-e156aef5-5f21-11e8-add9-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-567cd615-59a9-11e8-a847-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-0c030776-5e54-11e8-add9-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-216e9057-5a70-11e8-a847-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-117f2c0b-5f1b-11e8-add9-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-82d1de75-59b7-11e8-a847-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-c4929736-5a92-11e8-a6b4-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-f7886314-5f26-11e8-83b8-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-2d809486-59b8-11e8-a847-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-c97aad37-5e59-11e8-add9-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-6f5c2864-5a5d-11e8-a847-02b983f0a4db", "iqn.2016-09.com.openebs.jiva:pvc-360154c4-5f43-11e8-9b34-02b983f0a4db"]}

TASK [Check device nodes] *********************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:105
ok: [localhost] => {"ansible_facts": {"scsi_device": "/dev/sdc"}, "changed": false}

TASK [Create file system on iSCSI disk] *******************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:109
changed: [localhost -> None] => {"changed": true}

TASK [Mount device by Label] ******************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:117
changed: [localhost -> None] => {"changed": true, "dump": "0", "fstab": "/etc/fstab", "fstype": "ext4", "name": "/mnt/jiva", "opts": "discard,_netdev", "passno": "0", "src": "/dev/sdc"}

TASK [Place test data into volume] ************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:127
changed: [localhost -> None] => {"changed": true, "cmd": ["dd", "if=/dev/urandom", "of=/mnt/jiva/f1", "bs=1M", "count=1024"], "delta": "0:01:55.991792", "end": "2018-05-24 11:14:12.125790", "rc": 0, "start": "2018-05-24 11:12:16.133998", "stderr": "1024+0 records in\n1024+0 records out\n1073741824 bytes (1.1 GB, 1.0 GiB) copied, 115.952 s, 9.3 MB/s", "stderr_lines": ["1024+0 records in", "1024+0 records out", "1073741824 bytes (1.1 GB, 1.0 GiB) copied, 115.952 s, 9.3 MB/s"], "stdout": "", "stdout_lines": []}

TASK [Get the checksum of file created] *******************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:136
ok: [localhost -> None] => {"changed": false, "stat": {"atime": 1527160336.169304, "attr_flags": "e", "attributes": ["extents"], "block_size": 4096, "blocks": 2097160, "charset": "binary", "checksum": "252954732a48fc6ec5780726fdd45dd06c1218d7", "ctime": 1527160452.11925, "dev": 2080, "device_type": 0, "executable": false, "exists": true, "gid": 0, "gr_name": "root", "inode": 12, "isblk": false, "ischr": false, "isdir": false, "isfifo": false, "isgid": false, "islnk": false, "isreg": true, "issock": false, "isuid": false, "mimetype": "application/x-arc", "mode": "0644", "mtime": 1527160452.11925, "nlink": 1, "path": "/mnt/jiva/f1", "pw_name": "root", "readable": true, "rgrp": true, "roth": true, "rusr": true, "size": 1073741824, "uid": 0, "version": "18446744072863188497", "wgrp": false, "woth": false, "writeable": true, "wusr": true, "xgrp": false, "xoth": false, "xusr": false}}

TASK [Save checksum to variable] **************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:144
ok: [localhost] => {"ansible_facts": {"ckey": "252954732a48fc6ec5780726fdd45dd06c1218d7"}, "changed": false, "failed_when_result": false}

TASK [Unmount the volume] *********************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:149
changed: [localhost -> None] => {"changed": true, "dump": "0", "fstab": "/etc/fstab", "name": "/mnt/jiva", "opts": "defaults", "passno": "0"}

TASK [Create volume snapshot] *****************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:156
fatal: [localhost -> None]: FAILED! => {"changed": true, "cmd": "source ~/.profile; kubectl exec maya-apiserver-6d945596b-ff4ht -n openebs -c maya-apiserver -- mayactl snapshot create --volname pvc-360154c4-5f43-11e8-9b34-02b983f0a4db --snapname quicksnap", "delta": "0:00:00.821565", "end": "2018-05-24 11:14:18.972069", "failed_when_result": true, "msg": "non-zero return code", "rc": 1, "start": "2018-05-24 11:14:18.150504", "stderr": "Snapshot create failed: Volume 'pvc-360154c4-5f43-11e8-9b34-02b983f0a4db' not found\ncommand terminated with exit code 1", "stderr_lines": ["Snapshot create failed: Volume 'pvc-360154c4-5f43-11e8-9b34-02b983f0a4db' not found", "command terminated with exit code 1"], "stdout": "Executing volume snapshot create...", "stdout_lines": ["Executing volume snapshot create..."]}

TASK [Test Failed] ****************************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:268
ok: [localhost] => {"ansible_facts": {"flag": "Test Failed"}, "changed": false}

TASK [include_tasks] **************************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot-cleanup.yml:2
included: /home/vijay/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml for localhost

TASK [Deploy application files] ***************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml:2
changed: [localhost -> None] => (item=test-pvc.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f test-pvc.yaml -n simple-volume", "delta": "0:00:00.238211", "end": "2018-05-24 11:14:19.933595", "item": "test-pvc.yaml", "rc": 0, "start": "2018-05-24 11:14:19.695384", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"vut\" deleted", "stdout_lines": ["persistentvolumeclaim \"vut\" deleted"]}

TASK [Confirm that the storage volume is deleted] *********************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot-cleanup.yml:7
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pvc -n simple-volume", "delta": "0:00:00.236082", "end": "2018-05-24 11:14:20.716669", "rc": 0, "start": "2018-05-24 11:14:20.480587", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}

TASK [Test Cleanup Pass] **********************************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:277
ok: [localhost] => {"ansible_facts": {"cflag": "Cleanup Passed"}, "changed": false}

TASK [Terminate the log aggregator] ***********************************************************************************************************
task path: /home/vijay/openebs/e2e/ansible/playbooks/feature/snapshots/simple-volume/snapshot.yml:287
fatal: [localhost -> None]: FAILED! => {"changed": true, "cmd": "source ~/.profile; killall stern", "delta": "0:00:00.008453", "end": "2018-05-24 11:14:21.371599", "msg": "non-zero return code", "rc": 1, "start": "2018-05-24 11:14:21.363146", "stderr": "stern: no process found", "stderr_lines": ["stern: no process found"], "stdout": "", "stdout_lines": []}

PLAY RECAP ************************************************************************************************************************************
localhost                  : ok=30   changed=16   unreachable=0    failed=2   


```

vijaysinghrawat1 commented:
Done with the changes.
